src/bun.js/bindings/NodeHTTP.cpp

```diff
@@ -170,22 +169,38 @@ static void assignHeadersFromUWebSocketsForCall(uWS::HttpRequest* request, JSVal
                 setCookiesHeaderArray = constructEmptyArray(globalObject, nullptr);
                 RETURN_IF_EXCEPTION(scope, );
                 setCookiesHeaderString = nameString;
-                headersObject->putDirect(vm, nameIdentifier, setCookiesHeaderArray, 0);
+                headersMap.set(nameIdentifier.impl(), setCookiesHeaderArray);
                 RETURN_IF_EXCEPTION(scope, void());
             }
             arrayValues.append(setCookiesHeaderString);
             arrayValues.append(jsValue);
             setCookiesHeaderArray->push(globalObject, jsValue);
             RETURN_IF_EXCEPTION(scope, void());
-
         } else {
-            headersObject->putDirectMayBeIndex(globalObject, nameIdentifier, jsValue);
-            RETURN_IF_EXCEPTION(scope, void());
+            if (auto prev = headersMap.get(nameIdentifier.impl())) {
+                if (WTF::equal(nameIdentifier.impl(), "host"_s)) {
+                    continue;
+                }
+                headersMap.set(nameIdentifier.impl(), JSValue(jsString(vm, makeString(prev.getString(globalObject), ", "_s, jsValue))));
```

Comment:
- This is not safe for GC as the GC will not visit the JSValue created here.
- There is likely a faster way to do this.